py3status/modules/window_manipulation.py

```diff
@@ -0,0 +1,396 @@
+]
+
+old_default
+[
+    {'full_text': u'\u2756 Netflix '},
+    {'full_text': u'\u2756 #py3status - irssi'},
+    {'full_text': u'\u2756 z420: ~ '},
+    {'full_text': u'\u2756 ultrabug/py3status: py3statu '},
+    {'full_text': u'\u2756 window_manipulation.py (~/src/py3sta '},
+    {'full_text': u'\u2756 mpd_status.py (~/src/py3stat '},
+    {'full_text': u'\u2756 (10) Inbox - Firefox'},
+]
+
+colors
+[
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' Netflix '},
+    {'color': u'#FF48C4', 'full_text': u'\u2756'},
+    {'full_text': u' #py3status - irssi'},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' z420: ~ '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' ultrabug/py3status: py3statu '},
+    {'color': u'#FFAE3D', 'full_text': u'\u2756'},
+    {'full_text': u' window_manipulation.py (~/src/py3sta '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' mpd_status.py (~/src/py3stat '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' (10) Inbox - Firefox'},
+]
+
+"""
+
+from threading import Thread
+
+import i3ipc
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    allow_urgent = None
+    button_focus = None
+    button_next = None
+    button_previous = None
+    format = '{format_window}'
+    format_separator = ' '
+    format_window = '[\?if=is_focused&color=good \u2022|\?color=bad \u2022]'
+
+    def post_config_hook(self):
+        self.fmt_window = self.id = self.window_name = self.class_name = None
+
+        self.count_floating = self.count_scratchpad = \
+            self.count_urgent_sp = self.count_urgent = self.count_window = 0
+
+        t = Thread(target=self._loop)
+        t.daemon = True
+        t.start()
+
+    def _loop(self):
+        def update_window_manipulation(i3, e=None):
+            scratchpads = i3.get_tree().scratchpad().leaves()
+            windows = i3.get_tree().leaves()
+
+            # for w in windows:
+            #     print('OUTPUT... %s' % w.border)
+
+            new_scratchpad, new_window = self._organize_data(scratchpads, windows)
+            self._manipulate_and_refresh(new_scratchpad, new_window)
+
+        i3 = i3ipc.Connection()
+        update_window_manipulation(i3)
+
+        i3.on('window::new', update_window_manipulation)
+        i3.on('window::close', update_window_manipulation)
+        i3.on('window::focus', update_window_manipulation)
+        i3.on('window::title', update_window_manipulation)
+        i3.on('window::fullscreen_mode', update_window_manipulation)
+        i3.on('window::move', update_window_manipulation)
+        i3.on('window::floating', update_window_manipulation)
+        i3.on('window::urgent', update_window_manipulation)
+        i3.on('window::mark', update_window_manipulation)
+        i3.on('workspace::focus', update_window_manipulation)
+        i3.main()
```

Comment:
with this function i'd do
```
def _loop(self):
    try:
        ...
    except:
        self.py3.report_exception('monitor thread error')
```

otherwise we do not see the errors if they happen in the thread which is a pain to debug